<?xml version="1.0"?>
<robot name="virtualarm" xmlns:xacro="http://ros.org/wiki/xacro">

  <xacro:property name="scale" value="1.8" />

  <link name="virtualarm_TorsoCenter">
    <visual>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="virtualarm_RTorsoError" type="revolute">
    <parent link="virtualarm_TorsoCenter"/>
    <child link="virtualarm_RTorso"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_RTorso">
    <visual>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="virtualarm_ROmoplata" type="revolute">
    <parent link="virtualarm_RTorso"/>
    <child link="virtualarm_RShoulder"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_RShoulder">
    <inertial>
      <mass value="0.3125"/>
      <inertia ixx="0.000361198" ixy="1.98347e-06" ixz="6.35907e-07" iyy="0.000137229" iyz="5.68878e-06" izz="0.000399194"/>
      <origin rpy="0 0 0" xyz="0.00018 0.01856 0.00099"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RShoulderPitch.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RShoulderPitch_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_RShoulderRoll" type="revolute">
    <parent link="virtualarm_RShoulder"/>
    <child link="virtualarm_RBicep"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_RBicep">
    <inertial>
      <mass value="0.50527"/>
      <inertia ixx="0.000500872" ixy="3.55918e-05" ixz="8.51263e-05" iyy="0.00126284" iyz="-9.06813e-06" izz="0.00115718"/>
      <origin rpy="0 0 0" xyz="0.08151 -0.01284 -0.01017"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RShoulderRoll.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RShoulderRoll_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_RElbowYaw" type="revolute">
    <parent link="virtualarm_RBicep"/>
    <child link="virtualarm_RElbow"/>
    <origin rpy="0 -0.157079 0" xyz="0.1812 -0.015 0.00013"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="5.428" lower="-2.08567" upper="2.08567" velocity="7.33998"/>
  </joint>
  <link name="virtualarm_RElbow">
    <inertial>
      <mass value="0.27495"/>
      <inertia ixx="0.000122971" ixy="-9.86024e-07" ixz="-2.55838e-05" iyy="0.000331766" iyz="1.98925e-06" izz="0.000303047"/>
      <origin rpy="0 0 0" xyz="-0.02297 0.00019 0.00119"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RElbowYaw.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RElbowYaw_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_RElbowRoll" type="revolute">
    <parent link="virtualarm_RElbow"/>
    <child link="virtualarm_RForeArm"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="0.00872665" upper="1.56207" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_RForeArm">
    <inertial>
      <mass value="0.16579"/>
      <inertia ixx="0.00156513" ixy="-1.47517e-05" ixz="-6.05746e-05" iyy="0.00173219" iyz="1.7619e-07" izz="0.000273081"/>
      <origin rpy="0 0 0" xyz="0.0593 -0.00271 -0.01014"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RElbowRoll.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RElbowRoll_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_RWristYaw" type="revolute">
    <parent link="virtualarm_RForeArm"/>
    <child link="virtualarm_r_wrist"/>
    <origin rpy="0 0 0" xyz="0.15 0 0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="0.2014" lower="-1.82387" upper="1.82387" velocity="17.3835"/>
  </joint>
  <link name="virtualarm_r_wrist">
    <inertial>
      <mass value="0.27555"/>
      <inertia ixx="0.000151643" ixy="-1.06629e-05" ixz="2.89509e-06" iyy="0.000431099" iyz="3.13715e-06" izz="0.000456172"/>
      <origin rpy="0 0 0" xyz="0.03536 0.00072 -0.00207"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RWristYaw.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/RWristYaw_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_RHand" type="revolute">
    <parent link="virtualarm_r_wrist"/>
    <child link="virtualarm_r_gripper"/>
    <origin rpy="0 0 0" xyz="0.025 0 0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="0.144" lower="0.02" upper="0.98" velocity="12.56"/>
  </joint>
  <link name="virtualarm_r_gripper">
    <inertial>
      <mass value="2e-06"/>
      <inertia ixx="1.1e-09" ixy="0" ixz="0" iyy="1.1e-09" iyz="0" izz="1.1e-09"/>
    </inertial>
    <visual>
      <geometry>
        <cylinder radius="0.01" length="0.06"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.03"/>
    </visual>
  </link>

  <joint name="virtualarm_LTorsoError" type="revolute">
    <parent link="virtualarm_TorsoCenter"/>
    <child link="virtualarm_LTorso"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_LTorso">
    <visual>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="virtualarm_LOmoplata" type="revolute">
    <parent link="virtualarm_LTorso"/>
    <child link="virtualarm_LShoulder"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_LShoulder">
    <inertial>
      <mass value="0.3125"/>
      <inertia ixx="0.000361198" ixy="1.98347e-06" ixz="6.35907e-07" iyy="0.000137229" iyz="5.68878e-06" izz="0.000399194"/>
      <origin rpy="0 0 0" xyz="0.00018 0.01856 0.00099"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LShoulderPitch.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LShoulderPitch_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_LShoulderRoll" type="revolute">
    <parent link="virtualarm_LShoulder"/>
    <child link="virtualarm_LBicep"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_LBicep">
    <inertial>
      <mass value="0.50527"/>
      <inertia ixx="0.000500872" ixy="3.55918e-05" ixz="8.51263e-05" iyy="0.00126284" iyz="-9.06813e-06" izz="0.00115718"/>
      <origin rpy="0 0 0" xyz="0.08151 -0.01284 -0.01017"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LShoulderRoll.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LShoulderRoll_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_LElbowYaw" type="revolute">
    <parent link="virtualarm_LBicep"/>
    <child link="virtualarm_LElbow"/>
    <origin rpy="0 -0.157079 0" xyz="0.1812 -0.015 0.00013"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="5.428" lower="-2.08567" upper="2.08567" velocity="7.33998"/>
  </joint>
  <link name="virtualarm_LElbow">
    <inertial>
      <mass value="0.27495"/>
      <inertia ixx="0.000122971" ixy="-9.86024e-07" ixz="-2.55838e-05" iyy="0.000331766" iyz="1.98925e-06" izz="0.000303047"/>
      <origin rpy="0 0 0" xyz="-0.02297 0.00019 0.00119"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LElbowYaw.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LElbowYaw_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_LElbowRoll" type="revolute">
    <parent link="virtualarm_LElbow"/>
    <child link="virtualarm_LForeArm"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="0.00872665" upper="1.56207" velocity="9.22756"/>
  </joint>
  <link name="virtualarm_LForeArm">
    <inertial>
      <mass value="0.16579"/>
      <inertia ixx="0.00156513" ixy="-1.47517e-05" ixz="-6.05746e-05" iyy="0.00173219" iyz="1.7619e-07" izz="0.000273081"/>
      <origin rpy="0 0 0" xyz="0.0593 -0.00271 -0.01014"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LElbowRoll.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LElbowRoll_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_LWristYaw" type="revolute">
    <parent link="virtualarm_LForeArm"/>
    <child link="virtualarm_l_wrist"/>
    <origin rpy="0 0 0" xyz="0.15 0 0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="0.2014" lower="-1.82387" upper="1.82387" velocity="17.3835"/>
  </joint>
  <link name="virtualarm_l_wrist">
    <inertial>
      <mass value="0.27555"/>
      <inertia ixx="0.000151643" ixy="-1.06629e-05" ixz="2.89509e-06" iyy="0.000431099" iyz="3.13715e-06" izz="0.000456172"/>
      <origin rpy="0 0 0" xyz="0.03536 0.00072 -0.00207"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LWristYaw.dae" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://pepper_meshes/meshes/1.0/LWristYaw_0.10.stl" scale="${0.1*scale} ${0.1*scale} ${0.1*scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="virtualarm_LHand" type="revolute">
    <parent link="virtualarm_l_wrist"/>
    <child link="virtualarm_l_gripper"/>
    <origin rpy="0 0 0" xyz="0.025 0 0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="0.144" lower="0.02" upper="0.98" velocity="12.56"/>
  </joint>
  <link name="virtualarm_l_gripper">
    <inertial>
      <mass value="2e-06"/>
      <inertia ixx="1.1e-09" ixy="0" ixz="0" iyy="1.1e-09" iyz="0" izz="1.1e-09"/>
    </inertial>
    <visual>
      <geometry>
        <cylinder radius="0.01" length="0.06"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.03"/>
    </visual>
  </link>
</robot>

